### PR TITLE
Added redux boiler plate and intial setup

### PR DESCRIPTION
**Pull request by AakashMallik**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- [ ] Bug fix
- [x] Feature implementation
- [ ] Doc updates

* **What changes have you introduced?**
- Introduced redux as a global state management system in the project
- Snackbar, device connection status and dialogue box connected to redux store.

* **Does this PR introduce a breaking change?**
NO

### CODE REVIEW

File: src/redux/reducers/app.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+          message,
+          timeout,
+        },
+      };
+    }
+    case CLOSE_SNACKBAR: {
+      return {
+        ...prevState,
+        snackbar: {
+          ...initialState.snackbar,
```
Comment (CloudyPadmal) [inline src/redux/reducers/app.js]: no `isOpen: false`?

File: src/redux/reducers/app.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            ...deviceInformation,
+          },
+        },
+      };
+    }
+    case DEVICE_DISCONNECTED: {
+      return {
+        ...prevState,
+        device: {
+          ...initialState.device,
```
Comment (CloudyPadmal) [inline src/redux/reducers/app.js]: similarly here `isConnected: false`?

File: src/screen/PowerSource/PowerSource.js [hunk 3]
```diff
@@ -80,7 +80,7 @@ class PowerSouce extends Component {
 
   render() {
     const { pv1, pv2, pv3, pcs } = this.state;
-    const { onOpenDialog } = this.props;
+    // const { onOpenDialog } = this.props;
```
Comment (CloudyPadmal) [inline src/screen/PowerSource/PowerSource.js]: Not gonna need these anymore right?

File: src/redux/reducers/app.js (hunk 2 above)
Comment (AakashMallik) [reply src/redux/reducers/app.js]: initial state is **false**

File: src/screen/PowerSource/PowerSource.js (hunk 3 above)
Comment (AakashMallik) [reply src/screen/PowerSource/PowerSource.js]: addressed

**Pull request merged**